File: oms/src/main/java/us/avn/oms/domain/PlotGroup.java
/*******************************************************************************
 * Copyright (C) 2018 Laboratorio de Lobo Azul
 *  
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *  
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program.  If not, see <http://www.gnu.org/licenses/>.
 *******************************************************************************/
package us.avn.oms.domain;

import java.io.Serializable;


public class PlotGroup extends OMSObject implements Serializable {
	
	private static final long serialVersionUID = 8751282105532159742L;
	
	private Long id;
	private String name;
	private Long id1;
	private Long id2;
	private Long id3;
	private Long id4;
	private String active;
	private String source;
//    private Collection<IdName> aiList;
    
    public PlotGroup() {}

    public PlotGroup( Long i, String n ) {
    	this.id = i;
    	this.name = n;
    }
    	
	public Long getId() {
		return id;
	}

	public void setId(Long id) {
		this.id = id;
	}


	public String getName() {
		return name;
	}

	public void setName(String n) {
		this.name = n;
	}
		

	public Long getId1() {
		return id1;
	}

	public void setId1(Long id1) {
		this.id1 = id1;
	}


	public Long getId2() {
		return id2;
	}

	public void setId2(Long id2) {
		this.id2 = id2;
	}


	public Long getId3() {
		return id3;
	}

	public void setId3(Long id3) {
		this.id3 = id3;
	}


	public Long getId4() {
		return id4;
	}

	public void setId4(Long id4) {
		this.id4 = id4;
	}

/*
	public Collection<IdName> getAiList() {
		return aiList;
	}

	public void setAiList(Collection<IdName> aiList) {
		this.aiList = aiList;
	}
*/

	public String getActive() {
		return active;
	}

	public void setActive(String a) {
		this.active = a;
	}
		

	public String getSource() {
		return source;
	}

	public void setSource(String source) {
		this.source = source;
	}

}

File: ui/src/mainpage/pages/objects/PlotGroup.js
/*************************************************************************
 * PlotGroup.js
 * Copyright (C) 2018  Laboratorio de Lobo Azul
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program.  If not, see <http://www.gnu.org/licenses/>.
 ***********************************************************************/

/**
 * This object defines the plot group.  We provide active transfers as 
 * fake plot groups cause that might be interesting information, but
 * we need to distinguish between a real plot group and a fake.
 * 
 * i   = plot group ID
 * n   = plot group name
 * a   = active flag
 * id1 = ID for first tag
 * id2 = ID for second tag
 * id3 = ID for third tag
 * id4 = ID for fourth tag
 * aiList = the list of tags to return to the server to update the plot
 *          group object.
 * s   = source [PG (plot group) or X (transfer)]
 */
export function PlotGroup(i,n,a,id1,id2,id3,id4,s) { 
  this.id=i; this.name=n; this.active=a; this.id1=id1;
  this.id2=id2; this.id3=id3; this.id4=id4;
  this.aiList=null; this.source=s;
}

/**
 * This is an object that defines the "details" for a plot
 * n = number of days to request; the end point is always now
 * mx1 = max value for tag 1
 * mn1 = minimum value for tag 1
 * (etc)
 */
export function PlotDetails(n,mx0,mn0,mx1,mn1,mx2,mn2,mx3,mn3) {
  this.numberDays=n;
  this.max0=mx0; this.min0=mn0;
  this.max1=mx1; this.min1=mn1;
  this.max2=mx2; this.min2=mn2;
  this.max3=mx3; this.min3=mn3;
}
